janusgraph-core/src/main/java/org/janusgraph/graphdb/database/idassigner/placement/PropertyPlacementStrategy.java

```diff
@@ -89,8 +89,14 @@ private int getPartitionIDbyKey(JanusGraphVertex vertex) {
                 "PropertyPlacementStrategy has not been initialized correctly");
         assert idManager.getPartitionBound()<=Integer.MAX_VALUE;
         int partitionBound = (int)idManager.getPartitionBound();
-        final JanusGraphVertexProperty p = Iterables.getFirst(vertex.query().keys(key).properties(), null);
-        if (p==null) return -1;
+        Iterator<JanusGraphVertexProperty> it = vertex.query().keys(key).properties().iterator();
```

Comment:
I don't know a better way without using Guava here.
The only code I could think of, is assigning `null` to `p` instead of returning `-1` directly if there is nothing it the iterator.
Something like:
```
Iterator<JanusGraphVertexProperty> it = vertex.query().keys(key).properties().iterator();
final JanusGraphVertexProperty p = it.hasNext()? it.next() : null;
if (p==null){
    return -1;
}
```
I don't see much difference here. `p = it.hasNext()? it.next() : null;` is 1 operation more then checking `if(!it.hasNext()) return -1;` but a little bit cleaner.

What do you think about it?

Reply:
I like your new idea.

Reply:
Updated https://github.com/JanusGraph/janusgraph/compare/be12be0554c2e347a77c8a509656267d64dd4cd0..9792ca3d9a741e77f55edbb30ea4695c9d994c71